mods/seasonal/moves.js

```diff
@@ -572,6 +572,61 @@ let BattleMovedex = {
 		target: "self",
 		type: "Normal",
 	},
+	// Quite Quiet
+	"spookytransform": {
+		accuracy: true,
+		basePower: 0,
+		category: "Status",
+		desc: "",
+		shortDesc: "",
+		id: "spookytransform",
+		name: "Spooky Transform",
+		pp: 5,
+		priority: 1,
+		flags: {mystery: 1},
+		onPrepareHit: function (target, source) {
+			this.attrLastMove('[still]');
+			return !source.removeVolatile('spookytransform');
+		},
+		onHit: function (target, pokemon) {
+			if (!pokemon.transformInto(target, pokemon)) {
+				return false;
+			}
+			this.add('-anim', pokemon, "Refresh", pokemon);
+			pokemon.addVolatile('spookytransform');
+		},
+		effect: {
+			onDamagePriority: -100,
+			onDamage: function (damage, target, source) {
+				if (!target.transformed) return damage;
+				this.add('-immune', target, '[msg]', '[from] effect: Spooky Transform');
+				this.add('-message', `${target.name} reflected the damage back to ${source.name}.`);
+				this.add('-anim', target, "Night Shade", source);
+				this.damage(damage, source);
+				return null;
+			},
+			onResidualOrder: 100,
+			onResidualSubOrder: 1,
+			onResidual: function (pokemon) {
+				pokemon.clearVolatile();
```

Comment:
clearing volatiles will also clear the custom volatile that handles switch messages and your innate.